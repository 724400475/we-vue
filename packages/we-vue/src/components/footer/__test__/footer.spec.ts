import { shallowMount } from '@vue/test-utils'
import Footer from '../footer'

describe('footer', () => {
  test('create', () => {
    const wrapper = shallowMount(Footer, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('wv-footer')
    expect(wrapper.classes()).toContain('weui-footer')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('text', () => {
    const wrapper = shallowMount(Footer, {
      propsData: {
        text: 'test',
      },
    })

    expect(wrapper.contains('p.weui-footer__text')).toBeTruthy()
    expect(wrapper.find('p.weui-footer__text').text()).toBe('test')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
