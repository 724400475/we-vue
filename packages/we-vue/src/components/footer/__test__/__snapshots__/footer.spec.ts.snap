// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer create 1`] = `
<div class="weui-footer">
  <!---->
  <p class="weui-footer__text"></p>
</div>
`;

exports[`footer text 1`] = `
<div class="weui-footer">
  <!---->
  <p class="weui-footer__text">test</p>
</div>
`;
